flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/utils/PipelineTestOnYarnEnvironment.java

```diff
@@ -0,0 +1,327 @@
+        YARN_CONFIGURATION.setInt(YarnConfiguration.RESOURCEMANAGER_CONNECT_MAX_WAIT_MS, 5000);
+        YARN_CONFIGURATION.set(TEST_CLUSTER_NAME_KEY, "flink-yarn-tests-application");
+    }
+
+    @Before
+    public void setupYarnClient() throws Exception {
+        if (yarnClient == null) {
+            yarnClient = YarnClient.createYarnClient();
+            yarnClient.init(getYarnConfiguration());
+            yarnClient.start();
+        }
+    }
+
+    @After
+    public void shutdownYarnClient() {
+        yarnClient.stop();
+    }
+
+    @AfterClass
+    public static void teardown() {
+
+        if (yarnCluster != null) {
+            LOG.info("Stopping MiniYarn Cluster");
+            yarnCluster.stop();
+            yarnCluster = null;
+        }
+
+        // Unset FLINK_CONF_DIR, as it might change the behavior of other tests
+        Map<String, String> map = new HashMap<>(System.getenv());
+        map.remove(ConfigConstants.ENV_FLINK_CONF_DIR);
+        map.remove("YARN_CONF_DIR");
+        map.remove("IN_TESTS");
+        CommonTestUtils.setEnv(map);
+
+        if (yarnSiteXML != null) {
+            yarnSiteXML.delete();
+        }
+    }
+
+    protected static YarnConfiguration getYarnConfiguration() {
+        return YARN_CONFIGURATION;
+    }
+
+    public static void startMiniYARNCluster() {
+        try {
+            LOG.info("Starting up MiniYARNCluster");
+            if (yarnCluster == null) {
+                final String testName =
+                        YARN_CONFIGURATION.get(PipelineTestOnYarnEnvironment.TEST_CLUSTER_NAME_KEY);
+                yarnCluster =
+                        new MiniYARNCluster(
+                                testName == null ? "YarnTest_" + UUID.randomUUID() : testName,
+                                NUM_NODEMANAGERS,
+                                1,
+                                1);
+
+                yarnCluster.init(YARN_CONFIGURATION);
+                yarnCluster.start();
+            }
+
+            File targetTestClassesFolder = new File("target/test-classes");
+            writeYarnSiteConfigXML(YARN_CONFIGURATION, targetTestClassesFolder);
+
+            Map<String, String> map = new HashMap<String, String>(System.getenv());
+            map.put(
+                    "IN_TESTS",
+                    "yes we are in tests"); // see YarnClusterDescriptor() for more infos
+            map.put("YARN_CONF_DIR", targetTestClassesFolder.getAbsolutePath());
+            map.put("MAX_LOG_FILE_NUMBER", "10");
+            CommonTestUtils.setEnv(map);
+
+            assertThat(yarnCluster.getServiceState()).isEqualTo(Service.STATE.STARTED);
+            // wait for the nodeManagers to connect
+            while (!yarnCluster.waitForNodeManagersToConnect(500)) {
+                LOG.info("Waiting for Nodemanagers to connect");
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            LOG.error("setup failure", ex);
+            fail("");
```

Comment:
Could be simplified by  fail("setup failure", ex); here?

Reply:
@MOBIN-F  Did you fix this?

Reply:
Done